packages/runtime/container-runtime/src/opLifecycle/batchManager.ts

```diff
@@ -87,6 +108,45 @@ export class BatchManager {
             },
         };
     }
+
+    private checkReferenceSequenceNumber(message: BatchMessage) {
+        if (this.pendingBatch.length === 0 || message.referenceSequenceNumber === this.pendingBatch[0].referenceSequenceNumber) {
+            // The reference sequence numbers are stable
+            return;
+        }
+
+        const telemetryProperties = {
+            referenceSequenceNumber: this.pendingBatch[0].referenceSequenceNumber,
+            messageReferenceSequenceNumber: message.referenceSequenceNumber,
+            type: message.deserializedContent.type,
+            length: this.pendingBatch.length,
+            enableOpReentryCheck: this.options.enableOpReentryCheck === true,
+        };
+        const error = new GenericError(
```

Comment:
Should this be UsageError? Not sure, but it feels like it should be, as it's an error caused by user usage pattern.

Reply:
Good point